app/queries/comments/community_wellness_query.rb

```diff
@@ -0,0 +1,66 @@
+# This query will return an array of hashes that have the following structure:
+# [
+#   {
+#     "user_id"=>11,
+#     "serialized_weeks_ago"=>"1,16,20",
+#     "serialized_comment_counts"=>"2,2,1"
+#   },
+#   ...
+# ]
+#
+# Each result (hash) comes with `user_id` and the data required to check for the
+# Community Wellness Badge. These limitations consist of users that have at
+# least 2 comments without negative reactions (thumbsdown/vomit) per week.
+#
+# Note: Using `connection.execute("").to_a` instead of `.find_by_sql` because
+# ActiveRecord doesn't populate any other fields that aren't included in the raw
+# result set, so we don't get any added value from instantiating AR objects
+module Comments
+  class CommunityWellnessQuery
+    def self.call
+      ActiveRecord::Base.connection.execute(sql_query).to_a
+    end
+
+    def self.sql_query
+      # Based on the following Blazer query:
+      # https://dev.to/admin/blazer/queries/677
+      <<~SQL
+        SELECT user_id,
+             /* A comma separated string of "weeks_ago" */
+             array_to_string(array_agg(weeks_ago), ',') AS serialized_weeks_ago,
+             /* A comma separated string of comment counts.  The first value in this string happens on the week that is the first value in serialized_weeks_ago */
+             array_to_string(array_agg(number_of_comments_with_positive_reaction), ',') AS serialized_comment_counts
+        FROM
+        (SELECT user_id,
+             COUNT(user_id) as number_of_comments_with_positive_reaction,
+             /* Get the number of weeks, since today for posts */
+             (trunc((extract(epoch FROM (current_timestamp- created_at))) / 604800) + 1) AS weeks_ago
```

Comment:
FYI @jeremyf I added a `+1` to the truncate so indexing starts at `1` instead of `0`. This simplifies the logic in the service that processes these results.

I wasn't sure if `ceil` would work as a replacement here, in the specific case where the result is already a number without decimals (unlikely but possible?).

This one and removing the unserialized result column are the only changes made to the query. Since we're not instantiating ActiveRecord objects we don't gain anything from it, on top of some other details that weren't that helpful even with that in place (results arrived in scientific notation and had to be checked against integers).

Anyways, tests and tweaks will be coming soon before taking out of Draft 🤞